native/frontend/src/org/jetbrains/kotlin/resolve/konan/diagnostics/DefaultErrorMessagesNative.kt

```diff
@@ -38,7 +38,12 @@ private val DIAGNOSTIC_FACTORY_TO_RENDERER by lazy {
             "@ThreadLocal is applicable only to property with backing field, to property with delegation or to objects"
         )
         put(ErrorsNative.INAPPLICABLE_THREAD_LOCAL_TOP_LEVEL, "@ThreadLocal is applicable only to top level declarations")
-        put(ErrorsNative.INVALID_CHARACTERS_NATIVE, "Name {0}", CommonRenderers.STRING);
+        put(ErrorsNative.INVALID_CHARACTERS_NATIVE, "Name {0}", CommonRenderers.STRING)
+        put(ErrorsNative.INAPPLICABLE_OBJC_NAME, "@ObjCName is not applicable on overrides")
+        put(ErrorsNative.INVALID_OBJC_NAME, "@ObjCName should have a name and/or swiftName")
+        put(ErrorsNative.INVALID_CHARACTERS_OBJC_NAME, "@ObjCName contains illegal characters: \"{0}\"", CommonRenderers.STRING)
+        put(ErrorsNative.INCOMPATIBLE_OBJC_NAME_OVERRIDE, "Member inherits inconsistent @ObjCName")
+        put(ErrorsNative.INAPPLICABLE_EXACT_OBJC_Name, "Exact @ObjCName is only applicable to classes, objects and interfaces")
```

Comment:
> only applicable to classes, objects and interfaces

Is it applicable to enums? Same question for the annotation kdocs.

Reply:
Yes it is. Same goes for data classes. Both are considered classes, right? Or would it be preferable to explicitly mention them?

Reply:
> Or would it be preferable to explicitly mention them?

That was the original intention of my question, but

> Yes it is. Same goes for data classes. Both are considered classes, right?

You are right, `enum class` is `class`, so it doesn't make any sense to mention it in the error message. When writing the comment, I was more focused on the internal representation point of view -- there are different `ClassKind` values for classes and enums.